Integrate DFU target SMP (#12073)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

    * An overlay for using DTLS Connection Identifier.
      This significantly reduces the DTLS handshake overhead when doing the LwM2M Update operation.

>     * Support for Hosting MCUmgr client for external MCU. New overlay for enable feature and Device
>       tree overlays for UART1, UART2 and MCUBoot recovery mode.

Reviewer: ```suggestion
    * Support for hosting MCUmgr client in an external MCU.
      Including a new overlay file for enabling this and devicetree overlay files for UART1, UART2, and MCUBboot recovery mode.
```

Author: We need to remove that UART1. It is not valid

    * An overlay for using DTLS Connection Identifier.
      This significantly reduces the DTLS handshake overhead when doing the LwM2M Update operation.

>     * Support for Hosting MCUmgr client for external MCU. Including a new overlay file for enabling this and devicetree overlay files for UART2 and MCUBboot recovery mode.

Reviewer: ```suggestion
    * Support for Hosting MCUmgr client for external MCU.
      Including a new overlay file for enabling this and devicetree overlay files for UART2 and MCUboot recovery mode.
```

    * Support for using pre-provisioned X.509 certificates.
    * Support for using DTLS Connection Identifier

>     * Support for MCUmgr SMP client for external FOTA support with shell.

Reviewer: ```suggestion
    * Support for MCUmgr SMP client when using a shell command to perform a FOTA on an external SoC.
```
This need your careful attention. I could not really make heads or tails of this, but I tried.

Author: Support for MCUmgr SMP client to perform a FOTA on an external SoC

Reviewer: Sounds a lot better.

  * Updated:

    * Zephyr's LwM2M Connectivity Monitor object now uses a 16-bit value for radio signal strength so it does not roll over on values smaller than -126 dBm.

>     * LwM2M FOTA advanced FOTA support external MCU with DFU SMP target.
>     * Integrate new Fota download utils API for LwM2M firmware utils library.

Reviewer: ```suggestion
    * Advanced LwM2M FOTA support for an external MCU with DFU SMP target.
    * FOTA download Utils API integrated to the library.
```
Isn't the latter one rather something that is added (not updated), if the API is new?

Author: Ah true I will move these 2 lines to added

  * Updated:

    * The library now verifies whether the download started with the same URI and resumes the interrupted download.

>     * Extend library utils API which will support all fota DFU target's for download, shedule and activate image.

Reviewer: ```suggestion
    * Utils API to support downloading, scheduling and activating images in all FOTA DFU targets.
```
Not sure about this either.

>     * Library support DFU_TARGET_SMP with new Utils API.

Reviewer: ```suggestion
    * Support for DFU_TARGET_SMP with Utils API.
```
